lib/elixir/lib/module/checker.ex

```diff
@@ -1,16 +1,212 @@
 defmodule Module.Checker do
-  def verify(module_map, _binary) do
+  @moduledoc false
+
+  defmodule Coordinator do
+    @moduledoc false
+
+    def start_link(maps) do
+      ets = :ets.new(:checker_cache, [:set, :public])
+
+      Enum.each(maps, fn map ->
+        exports = definitions_to_exports(map.definitions)
+        deprecated = :maps.from_list(map.deprecated)
+        cache_module(ets, map.module, exports, deprecated)
+      end)
+
+      {:ok, server} = :gen_server.start_link(__MODULE__, [], [])
+      {server, ets}
+    end
+
+    def init([]) do
+      {:ok, %{waiting: %{}}}
+    end
+
+    def stop({server, ets}) do
+      :ets.delete(ets)
+      :gen_server.stop(server)
+    end
+
+    def handle_call({:lock, module}, from, %{waiting: waiting} = state) do
+      case :maps.get(module, waiting, nil) do
```

Comment:
Use pattern matching for maps. The API is more idiomatic and faster. In this case:

```elixir
case waiting do
  %{^module => froms} ->
```

You should also be able to use `put_in` to put nested values.